### DISCUSSION TITLE
How can we find the scope of installed packages?

### DISCUSSION THREAD

**Discussion by eabase** [microsoft/winget-cli#3239] [Q&A] [answered]
Assuming a package have been installed with `winget` with or without the `--scope` options.  

**How can we find what scope a packages has been installed in?**

**Comment by Trenly** [comment]
Winget uses the registry to determine which applications are installed and the scope. If the registry keys fall under `HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\`  or `HKLM:\SOFTWARE\WOW6432Node\Microsoft\Windows\CurrentVersion\Uninstall\` then it is a machine scope. If it falls under `HKCU:` instead of `HKLM:` then it is a user scope. 

There isn't currently a way to show the installed scope except to query the registry manually.

**Comment by mdanish-kh** [comment, marked as answer]
While there isn't a way to query the scope of a particular installed app, you can list all apps of a particular scope installed on your machine and search the app you're looking for in the output of the following commands

List all user-scoped apps installed on your machine
```
winget list --scope user
```

List all machine-scoped apps (Requires you to run the shell as admin)
```
winget list --scope machine
```

If you only want to see apps that are available via the winget source, you can filter them by appending `--source winget`

**Comment by eabase** [comment]
Although I marked the above as an accepted answer, it would have made much more sense to be able to use: `winget list --scope` and actually have the scope shown in a `scope` column.

**Reply by mdanish-kh** [reply]
cc @denelon if the issue:
- https://github.com/microsoft/winget-cli/issues/1155

may be updated with this request
